### PR TITLE
statistics: add sampling logger

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/50132

Problem Summary:

If table analysis fails or there are no tables to analyze, we need to avoid printing duplicate logs.

### What changed and how does it work?

We have added a sampling logger that prints only two same logs within a minute. However, please note that we cannot keep creating the logger repeatedly. It is necessary to keep it as a singleton so that we can track the same log counts with the same logger. Otherwise, the logger won't function as intended.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Comment by hi-rustin** [comment]
Tested locally:
```log
[2024/03/13 18:23:23.553 +08:00] [INFO] [refresher.go:124] ["No table to analyze"] [category=stats]
[2024/03/13 18:23:24.500 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/13 18:23:26.552 +08:00] [INFO] [refresher.go:124] ["No table to analyze"] [category=stats]
[2024/03/13 18:23:34.500 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/13 18:23:44.500 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/13 18:23:54.500 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/13 18:24:04.500 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/13 18:24:14.500 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/13 18:24:15.126 +08:00] [INFO] [manager.go:364] ["get owner"] ["owner info"="[log-backup] /tidb/br-stream/owner ownerManager 265674cb-6155-4005-aece-362f37fc0ab5"] ["owner key"=/tidb/br-stream/owner/22318e375664bef6] [ownerID=265674cb-6155-4005-aece-362f37fc0ab5] [op=none]
[2024/03/13 18:24:20.526 +08:00] [INFO] [owner_daemon.go:56] ["daemon became owner"] [id=265674cb-6155-4005-aece-362f37fc0ab5] [daemon-id=LogBackup::Advancer]
[2024/03/13 18:24:20.527 +08:00] [INFO] [advancer.go:503] ["Subscription handler spawned."] [category="log backup subscription manager"]
[2024/03/13 18:24:24.500 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/13 18:24:26.552 +08:00] [INFO] [refresher.go:124] ["No table to analyze"] [category=stats]
```

File: pkg/statistics/handle/logutil/logutil.go [hunk 1]
```diff
@@ -15,11 +15,29 @@
...
+	"go.uber.org/zap/zapcore"
 )
 
 // StatsLogger with category "stats" is used to log statistic related messages.
+// Do not use it to log the message that is not related to statistics.
 func StatsLogger() *zap.Logger {
 	return logutil.BgLogger().With(zap.String("category", "stats"))
 }
+
+// StatsSamplerLogger with category "stats" is used to log statistic related messages.
```
Comment (AilinKid) [inline pkg/statistics/handle/logutil/logutil.go:31]: you can add more comment about the difference about passing the *logger down and calling the logutil.StatsSamplerLogger() multi times.

the latter will create many children loggers out, while the former is just a kind implementation of singleton. We are going to do the sampling for all priority queue log information, so its necessary to use a singleton to collect them all them sampling them.

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

**Comment by hi-rustin** [comment]
Tested again:
```log
[2024/03/14 11:41:34.669 +08:00] [INFO] [refresher.go:119] ["No table to analyze"] [category=stats]
[2024/03/14 11:41:37.673 +08:00] [INFO] [refresher.go:119] ["No table to analyze"] [category=stats]
[2024/03/14 11:41:44.638 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/14 11:41:54.639 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/14 11:42:04.638 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/14 11:42:14.639 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/14 11:42:24.639 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/14 11:42:26.628 +08:00] [INFO] [manager.go:364] ["get owner"] ["owner info"="[log-backup] /tidb/br-stream/owner ownerManager d3c4000e-0c69-455c-8c1e-ec117bfcc989"] ["owner key"=/tidb/br-stream/owner/22318e3b0ce0e2f5] [ownerID=d3c4000e-0c69-455c-8c1e-ec117bfcc989] [op=none]
[2024/03/14 11:42:31.633 +08:00] [INFO] [owner_daemon.go:56] ["daemon became owner"] [id=d3c4000e-0c69-455c-8c1e-ec117bfcc989] [daemon-id=LogBackup::Advancer]
[2024/03/14 11:42:31.635 +08:00] [INFO] [advancer.go:503] ["Subscription handler spawned."] [category="log backup subscription manager"]
[2024/03/14 11:42:34.639 +08:00] [WARN] [cgmon.go:128] ["failed to get cgroup memory limit"] [error="failed to read memory cgroup from cgroups file: /proc/self/cgroup: open /proc/self/cgroup: no such file or directory"]
[2024/03/14 11:42:34.678 +08:00] [INFO] [refresher.go:119] ["No table to analyze"] [category=stats]
[2024/03/14 11:42:37.671 +08:00] [INFO] [refresher.go:119] ["No table to analyze"] [category=stats]
```

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by elsa0520 — APPROVED** [review]
LGTM

**Review by AilinKid — APPROVED** [review]

**Pull request merged**